packages/i18n/src/async-locale-data/use-async-intl-messages.ts

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+export type TMessageTranslations = {
+  [key: string]: string;
+};
+export type TMessagesAsync = (locale: string) => Promise<TMessageTranslations>;
+export type TState = {
+  isLoading: boolean;
+  messages?: TMessageTranslations;
+  error?: Error;
+};
+export type THookOptions = {
+  locale?: string;
+  loader: TMessagesAsync;
+};
+
+const initialState: TState = {
+  isLoading: true,
+  messages: undefined,
+  error: undefined,
+};
+
+// Low level hook to load messages for a specific locale. The loading is async
+// because it's assumed that the translation files are dynamically imported (code splitted).
+const useAsyncIntlMessages = ({ locale, loader }: THookOptions): TState => {
+  const [state, setState] = React.useState(initialState);
+
+  React.useEffect(() => {
+    let _isUnmounting = false;
```

Comment:
I think I do the same in flopflip and use a ref inside a hook. Like `const isUnmountedRef = useIsUnmounted()`. Can't recall I just remember it took longer than I thought to get right. I'll check later.

Reply:
I took this approach from reading this: https://stackoverflow.com/a/56443045